src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -74,6 +79,38 @@ class UptimeResultProcessor(ResultProcessor[CheckResult, UptimeSubscription]):
     def get_subscription_id(self, result: CheckResult) -> str:
         return result["subscription_id"]
 
+    def check_and_update_regions(self, subscription: UptimeSubscription):
+        # Run region checks and updates roughly once an hour
+        chance_to_run = subscription.interval_seconds / timedelta(hours=1).total_seconds()
+        if random.random() < chance_to_run:
+            subscription_region_slugs = {r.region_slug for r in subscription.regions.all()}
+            active_region_slugs = {c.slug for c in get_active_region_configs()}
+            if subscription_region_slugs != active_region_slugs:
+                new_region_slugs = active_region_slugs - subscription_region_slugs
+                removed_region_slugs = subscription_region_slugs - active_region_slugs
```

Comment:
This is basically for when we decide to deactivate a region right

Reply:
Yeah, exactly